src/main/java/com/direwolf20/buildinggadgets/common/items/gadgets/GadgetGeneric.java

```diff
@@ -135,6 +138,10 @@ public static boolean getFuzzy(ItemStack stack) {
         return NBTHelper.getOrNewTag(stack).getBoolean("fuzzy");
     }
 
+    public static LiquidInteractions.GadgetGeneric getFluidInteractionMode(ItemStack stack) {
+        return LiquidInteractions.GadgetGeneric.fromNBT(NBTHelper.getOrNewTag(stack).getByte("fluidInteractionMode"));
```

Comment:
Please pass the NBT down to the Enum. However the NBT key **must become a constant**! This is to allow easier future modifications, as it is almost impossible to find al the usages of an hardcoded value like that.